ui/app/src/components/WorkflowStateManagement/WorkflowStateManagement.tsx

```diff
@@ -89,6 +89,7 @@ export function WorkflowStateManagement(props: WorkflowStateManagementProps) {
   const { embedded, showAppsButton = !embedded, onSubmittingAndOrPendingChange } = props;
   const [fetching, setFetching] = useState(false);
   const [items, setItems] = useState<PagedArray<SandboxItem>>(null);
+  const [totalItems, setTotalItems] = useState(0);
```

Comment:
Feels a bit like slipping the error under the rug. Why is it trying to render things before they are set? Is it maybe just missing a `.?` somewhere?

Is a new piece of state really necessary?

```suggestion
  const [items, setItems] = useState<PagedArray<SandboxItem>>(() => Object.assign([], { total: 0 }));
```

Reply:
Make sure you're actually testing the 2 reproduction paths and understanding what's happening.